src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -2,28 +2,29 @@
 {
     using System;
     using System.Collections.Generic;
-    using Microsoft.ApplicationInsights.AspNetCore.Extensions;
     using Microsoft.ApplicationInsights;
     using Microsoft.ApplicationInsights.AspNetCore;
+    using Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners;
+    using Microsoft.ApplicationInsights.AspNetCore.Extensions;
     using Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers;
-    using Microsoft.ApplicationInsights.Channel;
-    using Microsoft.ApplicationInsights.DataContracts;
     using Microsoft.ApplicationInsights.Extensibility;
-    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.AspNetCore.Http;
     using Microsoft.Extensions.Configuration;
     using Microsoft.Extensions.Configuration.Memory;
-    using AspNetCore.Http;
-    using AspNetCore.Mvc.Infrastructure;
+    using Microsoft.Extensions.Options;
 
 #if NET451
-    using ApplicationInsights.DependencyCollector;
-    using ApplicationInsights.Extensibility.PerfCounterCollector;
-    using ApplicationInsights.Extensibility.PerfCounterCollector.QuickPulse;
-    using ApplicationInsights.WindowsServer.TelemetryChannel;
+    using Microsoft.ApplicationInsights.DependencyCollector;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector;
 #endif
 
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/> that allow adding Application Insights services to application.
+    /// </summary>
     public static class ApplicationInsightsExtensions
     {
+        private const string VersionKeyFromConfig = "version";
```

Comment:
I already did, `ApplicationInsightsServiceOptions` is constructed with entry assembly version as and `ApplicationVersion` but we are keeping compatibility with previous SDK and allow overriding via config entry.